src/controllers/settings.js

```diff
@@ -2,16 +2,191 @@
+    req.session.user,
+    email
+  )
+
+  await _sendVerificationEmail(unverifiedSubscriber.id)
+
+  // TODO: what should we return to the client?
+  return res.json('Resent the email')
+}
+
+function _checkForDuplicateEmail (sessionUser, email) {
+  email = email.toLowerCase()
+  if (email === sessionUser.primary_email.toLowerCase()) {
+    throw fluentError('user-add-duplicate-email')
+  }
+
+  for (const secondaryEmail of sessionUser.email_addresses) {
+    if (email === secondaryEmail.email.toLowerCase()) {
+      throw fluentError('user-add-duplicate-email')
+    }
+  }
+}
+
+async function removeEmail (req, res) {
+  const emailId = req.body.emailId
+  const sessionUser = req.user
+  const existingEmail = await getEmailById(emailId)
+
+  if (existingEmail.subscriber_id !== sessionUser.id) {
+    throw fluentError('error-not-subscribed')
+  }
+
+  removeOneSecondaryEmail(emailId)
+  res.redirect('/user/settings')
+}
+
+async function resendEmail (req, res) {
+  const emailId = req.body.emailId
+  const sessionUser = req.user
+  const existingEmail = await getUserEmails(sessionUser.id)
+
+  const filteredEmail = existingEmail.filter(
+    (a) => a.email === emailId && a.subscriber_id === sessionUser.id
+  )
+
+  if (!filteredEmail) {
+    throw fluentError('user-verify-token-error')
+  }
+
+  await _sendVerificationEmail(emailId)
+
+  // TODO: what should we return to the client?
+  return res.json('Resent the email')
+}
+
+async function _sendVerificationEmail (emailId) {
+  const unverifiedEmailAddressRecord = await resetUnverifiedEmailAddress(
+    emailId
+  )
+  const recipientEmail = unverifiedEmailAddressRecord.email
+  const data = {
+    recipientEmail,
+    ctaHref: getVerificationUrl(unverifiedEmailAddressRecord),
+    utmCampaign: 'email_verify',
+    unsubscribeUrl: getUnsubscribeUrl(
+      unverifiedEmailAddressRecord,
+      'account-verification-email'
+    ),
+    whichPartial: 'email_partials/email_verify',
+    heading: getMessage('email-verify-heading'),
+    subheading: getMessage('email-verify-subhead'),
+    partial: { name: 'verify' }
+  }
+  await sendEmail(
+    recipientEmail,
+    getMessage('email-subject-verify'),
+    getTemplate(data, verifyPartial(data))
+  )
+}
+
+async function verifyEmail (req, res) {
+  const token = req.query.token
+  await verifyEmailHash(token)
+
+  return res.redirect('/user/settings')
+}
+
+async function updateCommunicationOptions (req, res) {
+  try {
+    const sessionUser = req.user
+    // 0 = Send breach alerts to the email address found in brew breach.
+    // 1 = Send all breach alerts to user's primary email address.
+    const allEmailsToPrimary = Number(req.body.communicationOption) === 1
+    const updatedSubscriber = await setAllEmailsToPrimary(
+      sessionUser,
+      allEmailsToPrimary
+    )
+    req.session.user = updatedSubscriber
+
+    return res.json('Comm options updated')
+  } catch (ex) {
+    return res.json('Error updating comm options')
```

Comment:
might want to log the `ex`

Reply:
Per above, I think if we don't try to catch here, the error handler function registered in Express will do the right thing. I think this is a good pattern overall, it's better not to catch exceptions if we aren't sure we can handle them, and in this case the catch-all handler does exactly what you're proposing (log the message and return a standardized format JSON response)